### PR TITLE
[4.0] Plg parentassociations

### PR DESCRIPTION
**Pull request by AresRoxx10**
Pull Request for Issue # .

### Summary of Changes
Implement a color presentation in which one can see if the same articles are in the same state in different languages. The user has to manually check as articles that are not actually marked, adjust if necessary, and finally confirm as current.
If there are several articles in different languages with the same content, they can already be linked to each other in order to identify their belonging together. In order to simplify the above-mentioned functionality, a master-slave relationship has also been implemented between the articles. Specifically, this means that within an article group an article exists which is defined as the main article (master). All related articles should derive from the main article and from their slaves. In the application, all slaves are marked as unactual as soon as the master changes.

### Testing Instructions
1. Install at least 4 (e.g., De, ID, EN, FR, AF) languages
When you install a language, you must add the following values ​​in the language as in  C:\xampp1\htdocs\WPW3\administrator\language\en-GB\en-GB.com_associations.ini 
add the following values ​​in the appropriate language.
COM_ASSOCIATIONS_SAVE_APPROVED="Save Approved"
COM_ASSOCIATIONS_NOT_APPROVED_NOPARENT="reference is not the master"
COM_ASSOCIATIONS_MESSAGE_APPROVED="approved"
COM_ASSOCIATIONS_NOTCHILD="firstly save the Target Article 

2. Create  first article (DE)
3. Create second article (ID)
4. Create a third article (EN)
5. First article with second associate
6. Compare languages and press "Save Approved" as shown in Figure 1
7. First article with third article associate
8. In the Component Multilingual Associtation, select the first article as shown in Figure 2
9. Select third article, red exclamation point should appear according to main language (Figure 3)
10. If item is green, then in the info text Approved = 1 (see first picture).
11. If articles are yellow, then in the info text Approved = 0.
12. If articles are gray, then in the info text without Approved

### Expected result
Figure 1:
test3
Figure 2:
test
Figure 3:
test2

### CODE REVIEW

**Comment by brianteeman** [comment]
Thank you for your contribution. Please can you review it and resolve the following

1. Comments should be in english only (not french and german)
2. Code should not be commented out
3. Please resolve the conflicting files
4. Please look at the codestyle - specifically the indentations/tabs See https://developer.joomla.org/coding-standards.html

**Comment by richard67** [comment]
In addition to Brian's findings:
- Option names should not be in German here: [https://github.com/joomla/joomla-cms/pull/17881/files#diff-1f7598604d325c4356abf84f682138b6R30](https://github.com/joomla/joomla-cms/pull/17881/files#diff-1f7598604d325c4356abf84f682138b6R30)
- In Joomla core the author in the manifest xml file is "Joomla! Project" and not a person like it is here: [https://github.com/joomla/joomla-cms/pull/17881/files#diff-4a2052239b149da5a477637cddec85e5R5](https://github.com/joomla/joomla-cms/pull/17881/files#diff-4a2052239b149da5a477637cddec85e5R5)
(There are a few exception when things were taken from 3rd party creators like e.g. Akeeba, but these are rare exceptions).
=> Like this PR here is now it should not be accepted.

**Comment by christianboulbi** [comment]
Thank u for your feedback.

I have made some changes. can u please check it again ?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17881">issues.joomla.org/tracker/joomla-cms/17881</a>.</sub>

**Comment by brianteeman** [comment]
### Indentation issues
screenshotr09-26-29

### German comments
screenshotr09-26-50

### Conflicts
screenshotr09-27-24

**Comment by christianboulbi** [comment]
Thanks brianteeman.
Is it ok now?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17881">issues.joomla.org/tracker/joomla-cms/17881</a>.</sub>

**Comment by brianteeman** [comment]
My screenshots were just examples. I still see lots of the issues I first mentioned - github will show you all the issues - please review them all

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 1]
```diff
@@ -279,11 +279,173 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
...
+
+		// Create associated items list.
+		foreach ($languages as $langCode => $language)
+		{
+			if ($langCode == $itemLanguage)
+			{
+				continue;
+			}
+			$approved = null;
+			if(isset($items[$langCode])){
```
Comment (C-Lodder) [inline administrator/components/com_associations/Helper/AssociationsHelper.php]: `if(isset($items[$langCode])){` change to:
```
if (isset($items[$langCode]))
{
```

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 2]
```diff
@@ -279,11 +279,173 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
...
+					->from($db->quoteName('#__item_associations'))
+					->where($db->quoteName('id') . ' = ' . $db->quote($id));
+
+				$db->setQuery($query);
+				$approved = $db->loadResult();
+			}
+
+
+			// CASE: Associated and approved
+			if (($associationMode == 1) )
```
Comment (C-Lodder) [inline administrator/components/com_associations/Helper/AssociationsHelper.php]: remove space between `) )`

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 3]
```diff
@@ -279,11 +279,173 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
...
+			{
+				if(!isset($items[$langCode])){
```
Comment (C-Lodder) [inline administrator/components/com_associations/Helper/AssociationsHelper.php]: spacing and put curly brace on new line

File: plugins/content/articleassociations/articleassociations.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+<extension version="3.1" type="plugin" group="content">
+    <name>articleassociations</name>
+    <creationDate>24.08.2017</creationDate>
+    <author>Joomla! Project</author>
+    <authorEmail>christian.djombissi.wameni@mni.thm.de</authorEmail>
+    <authorUrl>http://localhost/joomla-cms-staging/joomla-cms-staging/</authorUrl>
+    <copyright>Copyright</copyright>
+    <license>GNU/GPL</license>
+    <version>1.0</version>
+    <description>Administrate die Relation between articles</description>
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.xml]: Change description to en-GB

File: plugins/content/articleassociations/articleassociations.xml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<extension version="3.1" type="plugin" group="content">
+    <name>articleassociations</name>
+    <creationDate>24.08.2017</creationDate>
+    <author>Joomla! Project</author>
+    <authorEmail>christian.djombissi.wameni@mni.thm.de</authorEmail>
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.xml]: Change email to `admin@joomla.org`

File: plugins/content/articleassociations/articleassociations.xml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    <authorUrl>http://localhost/joomla-cms-staging/joomla-cms-staging/</authorUrl>
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.xml]: Change URL to `www.joomla.org`

File: plugins/content/articleassociations/articleassociations.xml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    <copyright>Copyright</copyright>
+    <license>GNU/GPL</license>
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.xml]: Change license to `GNU General Public License version 2 or later; see LICENSE.txt`

File: plugins/content/articleassociations/articleassociations.xml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<extension version="3.1" type="plugin" group="content">
+    <name>articleassociations</name>
+    <creationDate>24.08.2017</creationDate>
+    <author>Joomla! Project</author>
+    <authorEmail>christian.djombissi.wameni@mni.thm.de</authorEmail>
+    <authorUrl>http://localhost/joomla-cms-staging/joomla-cms-staging/</authorUrl>
+    <copyright>Copyright</copyright>
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.xml]: Change copyright to `(C) 2005 - 2017 Open Source Matters. All rights reserved.`

File: plugins/content/articleassociations/articleassociations.php [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+	{
+		$query = $this->db->getQuery(true)
+			->select($this->db->quoteName(array('parentid')))
+			->from($this->db->quoteName('#__item_associations'))
+			->where($this->db->quoteName('id') . ' = ' . (int) $id);
+		$this->db->setQuery($query);
+		$parentid = $this->db->loadResult();
+
+		return (int) $parentid;
+
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: remove empty lines

File: plugins/content/articleassociations/articleassociations.php [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+						);
+
+						$conditions = array($this->db->quoteName('id') . ' =' . $this->db->quote($association_id));
+						$query->update($this->db->quoteName('#__item_associations'))->set($fields)->where($conditions);
+						$this->db->setQuery($query);
+						$res = $this->db->execute();
+
+
+					}
+
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: remove empty line

File: plugins/content/articleassociations/articleassociations.php [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+						$fields = array(
+							$this->db->quoteName('approved') . ' =0',
+							$this->db->quoteName('parentid') . ' =' . $this->db->quote($id)
+						);
+
+						$conditions = array($this->db->quoteName('id') . ' =' . $this->db->quote($association_id));
+						$query->update($this->db->quoteName('#__item_associations'))->set($fields)->where($conditions);
+						$this->db->setQuery($query);
+						$res = $this->db->execute();
+
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: remove empty lines

File: plugins/content/articleassociations/articleassociations.php [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+			$this->db->setQuery($query);
+			$list = $this->db->loadRowList();
+			foreach ($list as $res)
+			{
+				foreach ($res as $association_id)
+				{
+
+					if ($id != $association_id)
+					{
+
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: remove empty lines

File: plugins/content/articleassociations/articleassociations.php [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+			$query->select($this->db->quoteName(array('id')));
+			$query->from($this->db->quoteName('#__associations'));
+			$query->where($this->db->quoteName('key') . " = " . $this->db->quote($result));
+			$this->db->setQuery($query);
+			$list = $this->db->loadRowList();
+			foreach ($list as $res)
+			{
+				foreach ($res as $association_id)
+				{
+
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: remove empty line

File: plugins/content/articleassociations/articleassociations.php [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+		if ($result && strcmp($this->getParentId($id), "0") == 0)
+		{
+			$db    = JFactory::getDbo();
+			$query = $this->db->getQuery(true);
+			$query->select($this->db->quoteName(array('id')));
+			$query->from($this->db->quoteName('#__associations'));
+			$query->where($this->db->quoteName('key') . " = " . $this->db->quote($result));
+			$this->db->setQuery($query);
+			$list = $this->db->loadRowList();
+			foreach ($list as $res)
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php:134]: add new line before `foreach`

File: plugins/content/articleassociations/articleassociations.php [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+
+		// Select all records from the user profile table where key begins with "custom.".
+		// Order it by the ordering field.
+		$query->select($this->db->quoteName(array('key')));
+		$query->from($this->db->quoteName('#__associations'));
+		$query->where($this->db->quoteName('id') . " = " . $this->db->quote($id));
+		$this->db->setQuery($query);
+		$result = $this->db->loadResult();
+
+
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: remove empty line

File: plugins/content/articleassociations/articleassociations.php [hunk 16]
```diff
@@ -0,0 +1,168 @@
...
+	public $liste1, $parent;
+
+	public function __construct(&$subject, $config)
+	{
+		parent::__construct($subject, $config);
+	}
+
+	public function onContentAfterSave($context, $article, $isNew)
+	{
+
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: remove empty line

File: plugins/content/articleassociations/articleassociations.php [hunk 17]
```diff
@@ -0,0 +1,168 @@
...
+class plgContentarticleassociations extends JPlugin
+{
+	/**
+	 * A Database instance
+	 *
+	 * @var    JDatabaseDriver
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $db;
+	public $liste1, $parent;
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: Add comment blocks for each class variable

File: plugins/content/articleassociations/articleassociations.php [hunk 18]
```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * @version		$Id: articleassociations.php revision date lasteditedby $
+ * @package		Joomla
+ * @subpackage	Content
+ * @copyright	Copyright (C) 2005 - 2008 Open Source Matters. All rights reserved.
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: 2008 >> 2017

File: plugins/content/articleassociations/articleassociations.php [hunk 19]
```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * @version		$Id: articleassociations.php revision date lasteditedby $
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: Fix indentation on these lines. Should be 2 spaces minimum and all aligned.

File: installation/sql/mysql/joomla.sql [hunk 20]
```diff
@@ -2104,6 +2097,17 @@ CREATE TABLE IF NOT EXISTS `#__viewlevels` (
...
 --
+-- Table structure for table `#__item_associations`
+--
+
+CREATE TABLE IF NOT EXISTS `#__item_associations` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT 'Primary Key',
+  `parentid` int(10) NOT NULL DEFAULT 0,
+  `approved` tinyint(1) NOT NULL DEFAULT 0,
+  PRIMARY KEY (`id`)
+  )ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 DEFAULT COLLATE=utf8mb4_unicode_ci AUTO_INCREMENT=7;
```
Comment (C-Lodder) [inline installation/sql/mysql/joomla.sql]: `AUTO_INCREMENT=7` ?

File: administrator/components/com_associations/tmpl/associations/default.php [hunk 21]
```diff
@@ -26,129 +26,181 @@
 	2  => 'icon-archive',
 );
 
+/**
+ * "grey" => 0, not associated
+ * "green" => 1,  associated and current
+ * "yellow" => 2, associated and not current
+ */
+$associationMode = array(
+	"grey"   => 0,
```
Comment (C-Lodder) [inline administrator/components/com_associations/tmpl/associations/default.php]: replace double quotes with single quotes

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 22]
```diff
@@ -279,11 +279,173 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
...
+				}
+			}
+
+			// CASE: Not associated
+			if (($associationMode == 0) && isset($items[$langCode]))
+			{
+				unset($items[$langCode]);
+				continue;
+			}
+
```
Comment (C-Lodder) [inline administrator/components/com_associations/Helper/AssociationsHelper.php:381]: remove empty line

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 23]
```diff
@@ -279,11 +279,173 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
...
+				}
+
+			}
+
+			// CASE: Associated and not approved
+			if ($associationMode == 2)
+			{
+				if(!isset($items[$langCode])){
+					continue;
+				}
```
Comment (C-Lodder) [inline administrator/components/com_associations/Helper/AssociationsHelper.php]: add new line to separate `if` statements

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 24]
```diff
@@ -279,11 +279,173 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
...
+			}
+
+
+			// CASE: Associated and approved
+			if (($associationMode == 1) )
+			{
+				if(!isset($items[$langCode])){
+					continue;
+				}
+				if($approved == 0){
```
Comment (C-Lodder) [inline administrator/components/com_associations/Helper/AssociationsHelper.php]: - add new line to separate if statements 
- curly brace on new line
- `if(` >> `if (`

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 25]
```diff
@@ -279,11 +279,173 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
...
+				$id = $items[$langCode]['id'];
+				$query = $db->getQuery(true)
+					->select($db->quoteName('approved'))
+					->from($db->quoteName('#__item_associations'))
+					->where($db->quoteName('id') . ' = ' . $db->quote($id));
+
+				$db->setQuery($query);
+				$approved = $db->loadResult();
+			}
+
```
Comment (C-Lodder) [inline administrator/components/com_associations/Helper/AssociationsHelper.php]: remove empty line

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 26]
```diff
@@ -279,11 +279,173 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
...
+		$canCreate        = self::allowAdd($extensionName, $typeName);
+
+		// Create associated items list.
+		foreach ($languages as $langCode => $language)
+		{
+			if ($langCode == $itemLanguage)
+			{
+				continue;
+			}
+			$approved = null;
```
Comment (C-Lodder) [inline administrator/components/com_associations/Helper/AssociationsHelper.php]: new lines before and after `$approved = null;`

**Comment by C-Lodder** [comment]
And you have conflicts :)

File: plugins/content/articleassociations/articleassociations.xml [hunk 27]
```diff
@@ -8,7 +8,7 @@
     <copyright>Copyright</copyright>
     <license>GNU/GPL</license>
     <version>1.0</version>
-    <description>Administrate die Relation between articles</description>
+    <description>manage the parent relations beetween articles</description>
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.xml]: - Uppercase `M` on `manage`
- `beetween` >> `between`

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 28]
```diff
@@ -293,13 +293,13 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
 	public static function getAssociationHtmlList1($extensionName, $typeName, $itemId, $itemLanguage, $addLink = true, $associationMode)
 	{
 
-		// enthält alle Verbindungen von Item
+		// contains all associations from items
```
Comment (C-Lodder) [inline administrator/components/com_associations/Helper/AssociationsHelper.php]: Please start inline comments with an uppercase letter

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 29]
```diff
@@ -293,13 +293,13 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
...
 		// Get the associations list for this item.
 		$items = self::getAssociationList($extensionName, $typeName, $itemId);
 		//var_dump($items);
 		$titleFieldName = self::getTypeFieldName($extensionName, $typeName, 'title');
 
-		// Alle Sprachen die aktiviert sind
+		// all languages that are activated
```
Comment (C-Lodder) [inline administrator/components/com_associations/Helper/AssociationsHelper.php]: Please start inline comments with an uppercase letter

File: plugins/content/articleassociations/articleassociations.php [hunk 30]
```diff
@@ -41,14 +41,19 @@ public function onContentAfterSave($context, $article, $isNew)
 
 		// If not parent exists, add it
 		if ($this->getParentCount($id) !== 1)
-		{
+		{
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: remove trailing whitespace

File: plugins/content/articleassociations/articleassociations.php [hunk 31]
```diff
@@ -1,10 +1,10 @@
 <?php
 /**
- * @version		$Id: articleassociations.php revision date lasteditedby $
- * @package		Joomla
- * @subpackage	Content
- * @copyright	Copyright (C) 2005 - 2008 Open Source Matters. All rights reserved.
- * @license		GNU/GPL, see LICENSE.php
+ * @version        $Id: articleassociations.php revision date lasteditedby $
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: Please ensure these are aligned

File: plugins/content/articleassociations/articleassociations.php [hunk 32]
```diff
@@ -74,30 +79,28 @@ public function onContentAfterSaveAssociations($context, $article, $isNew)
 
 		// if slave article is edit
 		if ((int) $this->getParentId($id) !== 0)
-		{
+		{
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: remove trailing whitespace

File: plugins/content/articleassociations/articleassociations.php [hunk 33]
```diff
@@ -74,30 +79,28 @@ public function onContentAfterSaveAssociations($context, $article, $isNew)
...
 		// if master article is edit
 
-		$query->clear()
-			->select(array('#__item_associations.id'))
-			->from($this->db->quoteName('#__item_associations'))
-			->join('INNER', $this->db->quoteName('#__associations') . ' ON (' . $this->db->quoteName('#__associations.id') . ' = ' . $this->db->quoteName('#__item_associations.id') . ')')
-			->where($this->db->quoteName('#__associations.key') . ' ='  . $this->db->quote($result),'AND')
-			->where($this->db->quoteName('#__item_associations.parentid') . ' = 0');
+		$query->clear()->select(array('#__item_associations.id'))
+			           ->from($this->db->quoteName('#__item_associations'))->join('INNER', $this->db->quoteName('#__associations') . ' ON (' . $this->db->quoteName('#__associations.id') . ' = ' . $this->db->quoteName('#__item_associations.id') . ')')->where($this->db->quoteName('#__associations.key') . ' =' . $this->db->quote($result), 'AND')->where($this->db->quoteName('#__item_associations.parentid') . ' = 0');
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: Put query clauses on separate lines, as Travis will start moaning about line length

File: plugins/content/articleassociations/articleassociations.php [hunk 34]
```diff
@@ -74,30 +79,28 @@ public function onContentAfterSaveAssociations($context, $article, $isNew)
...
 		$this->db->setQuery($query);
 		$liste1 = $this->db->loadRowList();
 
 		if ($result && strcmp($this->getParentId($id), "0") == 0)
-		{
+		{
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: remove trailing whitespace

File: plugins/content/articleassociations/articleassociations.php [hunk 35]
```diff
@@ -106,21 +109,20 @@ public function onContentAfterSaveAssociations($context, $article, $isNew)
 			$this->db->setQuery($query);
 			$list = $this->db->loadRowList();
 			foreach ($list as $res)
-			{
+			{
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: remove trailing whitespace

File: plugins/content/articleassociations/articleassociations.php [hunk 36]
```diff
@@ -140,26 +142,20 @@ public function onContentAfterSaveAssociations($context, $article, $isNew)
 
 	public function getParentCount($id)
 	{
-		$query = $this->db->getQuery(true)
-			->select(COUNT($this->db->quoteName(array('parentid'))))
-			->from($this->db->quoteName('#__item_associations'))
-			->where($this->db->quoteName('id') . ' = ' . (int) $id);
+		$query = $this->db->getQuery(true)->select(COUNT($this->db->quoteName(array('parentid'))))
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: Put query clauses on separate lines

File: plugins/content/articleassociations/articleassociations.php [hunk 37]
```diff
@@ -140,26 +142,20 @@ public function onContentAfterSaveAssociations($context, $article, $isNew)
...
 	}
 
 
 	public function getParentId($id)
 	{
-		$query = $this->db->getQuery(true)
-			->select($this->db->quoteName(array('parentid')))
-			->from($this->db->quoteName('#__item_associations'))
-			->where($this->db->quoteName('id') . ' = ' . (int) $id);
+		$query = $this->db->getQuery(true)->select($this->db->quoteName(array('parentid')))
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: Put query clauses on separate lines

File: plugins/content/articleassociations/articleassociations.php [hunk 38]
```diff
@@ -1,9 +1,9 @@
 <?php
 /**
- * @version        $Id: articleassociations.php revision date lasteditedby $
+ * @version        $Id:  articleassociations.php  revision  date  lasteditedby  $
```
Comment (C-Lodder) [inline plugins/content/articleassociations/articleassociations.php]: Now aligned, but they should only be separate from `@version`, etc, by 2 spaces. See: https://developer.joomla.org/coding-standards/docblocks.html

Comment (C-Lodder) [reply plugins/content/articleassociations/articleassociations.php]: Here's the updated version with some other tweaks: https://pastebin.com/BHCJ6ALE

**Comment by christianboulbi** [comment]
i think, it should be ok now<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17881">issues.joomla.org/tracker/joomla-cms/17881</a>.</sub>

**Comment by C-Lodder** [comment]
@christianboulbi - No, there are still loads of code style issues. Refer to the docs: https://developer.joomla.org/coding-standards/php-code.html

and you have conflicts, so you'll need to fix them

File: installation/sql/mysql/joomla.sql [hunk 39]
```diff
@@ -2113,3 +2111,15 @@ INSERT INTO `#__viewlevels` (`id`, `title`, `ordering`, `rules`) VALUES
...
+--
+-- Table structure for table `#__item_associations`
+--
+
+CREATE TABLE IF NOT EXISTS `#__item_associations` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT 'Primary Key',
+  `parentid` int(10) NOT NULL DEFAULT 0,
+  `approved` tinyint(1) NOT NULL DEFAULT 0,
+  PRIMARY KEY (`id`)
+  )ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 DEFAULT COLLATE=utf8mb4_unicode_ci;
```
Comment (infograf768) [inline installation/sql/mysql/joomla.sql:2125]: Error
Missing `COMMENT`
```
CREATE TABLE IF NOT EXISTS `#__item_associations` (
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT COMMENT 'Primary Key',
  `parentid` int(10) NOT NULL DEFAULT 0,
  `approved` tinyint(1) NOT NULL DEFAULT 0,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 DEFAULT COLLATE=utf8mb4_unicode_ci;
  
```

File: administrator/components/com_associations/Model/Association.php [hunk 40]
```diff
@@ -31,9 +31,162 @@ class Association extends ListModel
...
+    public function approved($id, $targetId)
+    {
+        if (!empty($targetId))
+        {
+            $parentId=0;
+            $db = $this->getDbo();
+            $selectQuery = $db->getQuery(true);
+            $selectQuery
+                ->select('id,parentid')
+                ->from($db->quoteName('item_associations'))
```
Comment (infograf768) [inline administrator/components/com_associations/Model/Association.php]: Error
Should be
`>from($db->quoteName('#__item_associations'))`
And also below a few times.

**Comment by infograf768** [comment]
Apart from the obvious errors I found (see above) and a lot of coding standards corrections to do (+ postgresql impossible to merge), + a few glitches (Target article Checked out after Save Approved) I am not sure this approach is the one we need.

Basically, and whatever the way the UI is displayed, I think we must always have a fixed reference language (Parent should be in the refernce language) and the possibility to filter stuff by outdated, not associated or up-to-date.

The problem is when you change languages.
Below, Article English has been modified in the Articles usual UI.
Therefore the FR icon displays as orange with NOT approved.

screen shot 2017-09-07 at 08 46 06

But, if I switch to French, I get
screen shot 2017-09-07 at 08 49 31

This is really misleading and therefore the reason why I insist on choosing always the same reference language.

I guess the new plugin is used to mark as Not approved the associated article when the "Parent" is modified. Unhappily, it is specific to articles, i.e. any other possible association (categories, menu items, contacts, etc. are not concerned. 

Please read this:
https://github.com/joomla/joomla-cms/pull/17802#issuecomment-326523579 to further understand my comment.

As you can see there, my feeling the only way to deal with this is dates/comparison.
The use of a new table as you do here could be the way to go.

**Comment by perrez** [comment]
infograf768. Thank u for your feedback.
if you change a master article, all slave-article of this article still need to be approved.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17881">issues.joomla.org/tracker/joomla-cms/17881</a>.</sub>

**Comment by infograf768** [comment]
The sql CREATE TABLE has still not been updated.

@perrez 
> if you change a master article, all slave-articles of this article still need to be approved.

That is not my point. Obviously, whether you call it approved or up-to-date the new status should be easy to find and act accordingly, the issue is that one has to have only ONE master article (or other type of item). 
IN the existng UI, switching languages will change the Master. We do not want that. We need to set a specific language as MASTER for the site in order to be able to get a usable feature (with filters, etc.). Please re-read my proposal in the other PR and both screenshots above where the problem displays perfectly.

We should not have to click on "Save Approved", but just saving the Target would make it automatically Up-to-date.

This means that if we have a new table to store correctly the modified date, and add this for every component when Assoc is on, Multilingual associations will know by date comparison if the target is Up-to-date or not.

Example:
I have an article tagged to English and the `Default site Reference language` is English (as proposed in the other PR).
The language filter plugin is set to use associations.
When saving this article, it will not only save it in the `content` table but also save its `id`, `type` and `modified date` in the new table (can be `item_associations` if you wish).
As the current Multingual Associations main page is not set for this — one can choose a reference in ANY language — we would have a new page ( the STATUS page) specially aimed at comparing THE MASTER to its SLAVES, letting filter the slaves per language used in the site or all languages (obvioulsy without the refernce language) and status.
It will then be real easy to know what is or not Up-to-date, etc.

On the Face to Face page (same one or a new one), once decided or not to change the Slave (or also maybe the Reference too), Saving Target would also Save the Reference and enter in the new Table the same Modified date, therefore updating the STATUS page where we compare MASTER to SLAVES when we go back there.

**Comment by perrez** [comment]
@infograf768 
Thanks for the proposal of  adding another Attribute to our table that will save the initial Article which has been changed and serves as Master. That is a good idea .
But our aim is to give the user the possibility to be able to manually check (approve) the Slave Article.  The Master article is the article without a parent article and it can derive the children article

As a  extension, we  will want that the Slave Article which has already been checked  should also become  a  master article.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17881">issues.joomla.org/tracker/joomla-cms/17881</a>.</sub>

**Comment by infograf768** [comment]
Sorry to disagree. If you are not implementing a specific language as master, this will be totally  confusing.

**Comment by jreys** [comment]
Yup, agree with @infograf768, without setting a master language and comparing modified dates, this can cause some troubles, his solution looks the most elegant for this problem

**Comment by brianteeman** [comment]
Going to close this one as after a year without update it has clearly been abandoned. It can always be reopened if updated.

**Pull request closed without merging**